core/client/src/main/java/alluxio/client/ClientContext.java

```diff
@@ -42,8 +41,12 @@
    * This method requires that configuration has been initialized.
    */
   public static void init() {
-    String masterHostname =
-        Preconditions.checkNotNull(Configuration.get(PropertyKey.MASTER_HOSTNAME));
+    String masterHostname;
+    if (Configuration.containsKey(PropertyKey.MASTER_HOSTNAME)) {
+      masterHostname = Configuration.get(PropertyKey.MASTER_HOSTNAME);
+    } else {
+      masterHostname = NetworkAddressUtils.getLocalHostName();
```

Comment:
localhost is a always where the master is running from the master's perspective, but rarely where the master is running from the worker's perspective. When the user is running a service which needs the master address, we want to force them to manually set it instead of defaulting to localhost.